File: 13-fibonacci/fibo.ml
(* Ocaml program to generate fibonacci numbers *)
(* Written by M.V.Harish Kumar on 29-05-2024 *)
let fib n =
  let rec fib_tr n a1 a2 =
    match n with
    | 0 -> a1
    | n -> fib_tr (n-1) (a1+a2) a1
  in fib_tr n 0 1

let () =
  print_endline "Fibonacci series printer\n";
  print_string "Enter no. of terms: ";
  let n = read_int () in
  Printf.printf "The first %d fibonacci terms are: \n\n" n;
  for i = 0 to n-1 do
    print_int (fib i);
    print_newline ()
  done

File: 13-fibonacci/fibo.go
// Fibonacci series printer in Go
// Written by M.V.Harish Kumar on 18-06-2025
package main

import "fmt"

func main() {
	var limit uint

	fmt.Println("Fibonacci series printer\n")
	fmt.Print("Enter no of terms to print: ")
	fmt.Scanln(&limit)

	a, b := 0, 1
	fmt.Printf("The first %d terms of fibonacci series are:\n", limit)
	for _ = range limit {
		fmt.Println(a)
		a, b = b, a+b
	}
}
